tests/functional/spec/timeout_modal.spec.js

```diff
@@ -0,0 +1,62 @@
+import TimeoutInterstitialPage from "../generated_pages/timeout_modal/timeout-modal-interstitial.page";
+import TimeoutSubmitPage from "../generated_pages/timeout_modal/submit.page";
+import { TimeoutModalPage } from "../base_pages/timeout-modal.page.js";
+import ThankYouPage from "../base_pages/thank-you.page.js";
+
+describe("Timeout Modal", () => {
+  describe("Given I am completing the survey, ", () => {
+    beforeEach(() => {
+      browser.openQuestionnaire("test_timeout_modal.json");
+    });
+    testSet(TimeoutInterstitialPage);
+  });
+});
+
+describe("Timeout Modal Post Submission", () => {
+  describe("Given I am completing the survey and get to post submission page, ", () => {
+    beforeEach(() => {
+      browser.openQuestionnaire("test_timeout_modal.json");
+      $(TimeoutInterstitialPage.submit()).click();
+      $(TimeoutSubmitPage.submit()).click();
+    });
+    testSet(ThankYouPage);
+  });
+});
+
+function testSet(page) {
+  it("When the session timeout is set to 125 seconds, Then it will make the timeout modal with the option to extend the session visible after 65 seconds and it will redirect after session expires", () => {
```

Comment:
The timeout seconds is not important, and this is specially about the redirect, so maybe it should be:
- When the timeout modal is displayed, and I do not extend my session
- Then I will be redirected to the session expired page

Reply:
Done, thanks!